### PR TITLE
More consistent readme for msrv

### PR DESCRIPTION
**Pull request by cijothomas**
Towards https://github.com/open-telemetry/opentelemetry-rust/issues/1306
This PR fixes some inconsistencies, but not all. There'll be further follow ups to make everything consistent.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,11 +17,10 @@ analysis in order to understand your software's performance and behavior. You
 can export and analyze them using [Prometheus], [Jaeger], and other
 observability tools.
 
-*Compiler support: [requires `rustc` 1.70+][msrv]*
+*[Supported Rust Versions](#supported-rust-versions)*
```
Comment (lalitb) [inline README.md:20]: This [section](https://github.com/open-telemetry/opentelemetry-rust?tab=readme-ov-file#supported-rust-versions) needs some changes, the MSRV varies across crates—some require Rust 1.70, while others need 1.71.

File: opentelemetry-appender-log/src/lib.rs [hunk 2]
```diff
@@ -93,6 +95,20 @@
...
+//! supported version is 1.70. The current OpenTelemetry version is not
+//! guaranteed to build on Rust versions earlier than the minimum supported
+//! version.
+//!
+//! The current stable Rust compiler and the three most recent minor versions
+//! before it will always be supported. For example, if the current stable
+//! compiler version is 1.49, the minimum supported version will not be
+//! increased past 1.46, three minor versions prior. Increasing the minimum
+//! supported compiler version is not considered a semver breaking change as
+//! long as doing so complies with this policy.
```
Comment (lalitb) [inline opentelemetry-appender-log/src/lib.rs:111]: should this be in README.md instead of lib.rs?

File: opentelemetry-appender-tracing/README.md [hunk 3]
```diff
@@ -32,3 +32,18 @@ management, and export of telemetry. A major goal of OpenTelemetry is that you
 can easily instrument your applications or systems, no matter their language,
 infrastructure, or runtime environment. Crucially, the storage and visualization
 of telemetry is intentionally left to other tools.
+
+*[Supported Rust Versions](#supported-rust-versions)*
```
Comment (lalitb) [inline opentelemetry-appender-tracing/README.md:36]: nit - do we need this line, given that the information is already clear and easy to find in the README?

Comment (cijothomas) [reply opentelemetry-appender-tracing/README.md:36]: kept it for consistency, and also there'll be more content to be added to this readme in future.

**Review by lalitb — APPROVED** [review]

**Pull request merged**